include/sys/avl_impl.h

```diff
@@ -147,7 +147,9 @@ struct avl_tree {
 	int (*avl_compar)(const void *, const void *);
 	size_t avl_offset;		/* offsetof(type, avl_link_t field) */
 	ulong_t avl_numnodes;		/* number of nodes in the tree */
+#ifndef _KERNEL
 	size_t avl_size;		/* sizeof user type struct */
+#endif
```

Comment:
please add a comment explaining why we have this in userland but not kernel. (backwards ABI compatibility for userland, performance for kernel)

Reply:
Added a comment and renamed the field to avl_pad to explicitly prevent any legacy use.